hook.js

```diff
@@ -39,16 +35,36 @@ function handle(request, response) {
     }
 }
 
-// TODO: Remove the commentURL parameter once parseCSS is refactored
-function addPullRequestComments(destinationRepo, originRepo, originBranch, prNumber, commentURL) {
+function processPullRequest(destinationRepo, originRepo, originBranch, prNumber, commentURL) {
+    var commits = getPullRequestCommits(destinationRepo, prNumber);
     var config = getConfig(originRepo, originBranch);
-    var pullRequestCommits = getPullRequestCommits(destinationRepo, prNumber);
 
-    Q.all([config, pullRequestCommits]).spread(function(config, pullRequestCommits) {
-        pullRequestCommits.forEach(function(currentCommit) {
-            getCommitDetail(originRepo, currentCommit.sha, function(error, currentCommitDetail) {
-                if (error) return logger.error('getCommitDetail failed:', error);
-                parseCSS(currentCommitDetail.files, config, commentURL, token, function(usageInfo) {}, currentCommit.sha);
+    // Once we pull down commit metadata and configuration...
+    Q.all([commits, config]).spread(function(commits, config) {
+
+        // Go through each commit...
+        commits.forEach(function(currentCommit) {
+
+            // And each file of each commit...
+            currentCommit.files.forEach(function(file) {
+
+                // And process the files
+
+                function handleIncompatibility(incompatibility) {
+                    var line = diff.lineToIndex(file.patch, incompatibility.usage.source.start.line);
+                    var comment = incompatibility.featureData.title + ' not supported by: ' + incompatibility.featureData.missing;
+                    commenter.postPullRequestComment(commentURL, comment, file.filename, currentCommit.sha, line);
+                }
+
+                switch (path.extname(file.filename).toLowerCase()) {
```

Comment:
Done! :smile:

This thread is still here because a different line was changed.